compiler-core/src/manifest.rs

```diff
@@ -333,3 +212,171 @@ where
+                    version: Version::new(1, 2, 3),
+                    build_tools: ["gleam".into()].into(),
+                    otp_app: None,
+                    requirements: vec![],
+                    source: ManifestPackageSource::Git {
+                        repo: "https://github.com/gleam-lang/gleam.git".into(),
+                        commit: "bd9fe02f72250e6a136967917bcb1bdccaffa3c8".into(),
+                    },
+                },
+                ManifestPackage {
+                    name: "awsome_local1".into(),
+                    version: Version::new(1, 2, 3),
+                    build_tools: ["gleam".into()].into(),
+                    otp_app: None,
+                    requirements: vec![],
+                    source: ManifestPackageSource::Local {
+                        path: PACKAGE.into(),
+                    },
+                },
+                ManifestPackage {
+                    name: "gleeunit".into(),
+                    version: Version::new(0, 4, 0),
+                    build_tools: ["gleam".into()].into(),
+                    otp_app: None,
+                    requirements: vec!["gleam_stdlib".into()],
+                    source: ManifestPackageSource::Hex {
+                        outer_checksum: Base16Checksum(vec![3, 46]),
+                    },
+                },
+            ],
+        };
+
+        let buffer = manifest.to_toml(TestMakeRelative {
+            curr_dir: HOME.into(),
+        });
+        assert_eq!(
+            buffer,
+            r#"# This file was generated by Gleam
+# You typically do not need to edit this file
+
+packages = [
+  { name = "aaa", version = "0.4.0", build_tools = ["rebar3", "make"], requirements = ["zzz", "gleam_stdlib"], otp_app = "aaa_app", source = "hex", outer_checksum = "0316" },
+  { name = "awsome_local1", version = "1.2.3", build_tools = ["gleam"], requirements = [], source = "local", path = "../path/to/package" },
+  { name = "awsome_local2", version = "1.2.3", build_tools = ["gleam"], requirements = [], source = "git", repo = "https://github.com/gleam-lang/gleam.git", commit = "bd9fe02f72250e6a136967917bcb1bdccaffa3c8" },
+  { name = "gleam_stdlib", version = "0.17.1", build_tools = ["gleam"], requirements = [], source = "hex", outer_checksum = "0116" },
+  { name = "gleeunit", version = "0.4.0", build_tools = ["gleam"], requirements = ["gleam_stdlib"], source = "hex", outer_checksum = "032E" },
+  { name = "zzz", version = "0.4.0", build_tools = ["mix"], requirements = [], source = "hex", outer_checksum = "0316" },
+]
+
+[requirements]
+aaa = { version = "> 0.0.0" }
+awsome_local1 = { path = "../path/to/package" }
+awsome_local2 = { git = "https://github.com/gleam-lang/gleam.git" }
+gleam_stdlib = { version = "~> 0.17" }
+gleeunit = { version = "~> 0.1" }
+zzz = { version = "> 0.0.0" }
+"#
+        );
+
+        // The deserialized path no longer matches because it contains a relative path for the
```

Comment:
What does this mean?

Reply:
For this part of the test, we were testing that when you run `manifest.to_toml`, and the deserialize that back to a `Manifest` struct, you get the same result. That is no longer the expected behavior, because `manifest.to_toml` will now possibly convert an absolute path to a relative path.
I don't love it, but I think it's okay. It would be different if this was the serialize/deserialize impl or something. I left it commented out so you would see it, but if you're okay with it I'll remove the commented code, if not we'll find another way to do it.